### DISCUSSION TITLE
LDAP Groups Refreshed For Token?

### DISCUSSION THREAD

**Discussion by tjmoyes** [dexidp/dex#2048] [Q&A] [answered]
Hey all,

I apologize if this is a dumb question but it has just crossed my mind, and I apologize if this is the wrong place to ask. 

So I've set up Dex/Gangway to authenticate into a Kube cluster with LDAP credentials. I've also set up a couple of Roles & RoleBindings based off groups in said LDAP. 

My question revolves around the refresh process: 

Let's say, when a user initially logs into Dex and gets the KubeConfig from Gangway, they are a part of Group A. At some point, they are removed from Group A and placed into Group B. Groups A and B have separate role bindings in the cluster. Does the refresh also refresh the groups that a user is a part of?

**Comment by nabokihms** [comment, marked as answer]
Hello, @tjmoyes.
Yes, it works as you described. If a connector implements the `RefreshConnector` interface, the `Refresh` method will be called on token refreshing. 

The LDAP refresh method looks like this https://github.com/dexidp/dex/blob/83ad7bc4e3beaf21ac3a56a03dd4e223249215fe/connector/ldap/ldap.go#L529-L568
So it does update the groups on line 561.

This is technical stuff, but I think it may be useful to understand how dex interacts with other identity providers.

**Reply by tjmoyes** [reply]
This is phenomenal! I still have to learn how Go works but I can at least understand what the code here is doing. 

Thank you so much!
